Adds query for user

gemcfadyen opened the pull request:
@Maikon : This pull request adds the functionality for querying a particular user.
If you have any feedback please comment on the PR and I will work it in.

Outstanding:
- Removing duplication between TransformUserResponse and TransformRepoResponse
- I've not tried deploying it yet, will do this in the morning.

NB: All out of hits on the real github api for today

gemcfadyen committed:
Adds the ability to query for a particular user

gemcfadyen committed:
Use pattern matching in FetchUser

gemcfadyen committed:
Adds test to ensure the body of the user response is json

gemcfadyen commented on web/github/transform_user_response.ex:
```diff
@@ -0,0 +1,28 @@
+    |> convert_map_to_struct
+  end
+
+  defp downcase_user(user) do
```
Most of the private functions are the same as in `TransformRepoResponse` so am thinking of extracting them out into some sort of response transformer that both can use.

Maikon replied:
Yep 👍

Maikon commented on web/github/fetch_user.ex:
```diff
@@ -0,0 +1,24 @@
+    transform_user(user, found_user?(user))
+  end
+
+  defp transform_user(user, true), do: user
```
We can pattern-match on the return value of `Storage.find_user(name)` directly here instead and that would remove the `found_user?` function.

`user -> user` , `nil -> fetch_from_api`

Maikon commented on test/controllers/github_controller_test.exs:
```diff
@@ -1,10 +1,27 @@
+
+    response = GithubService.Router.call(conn, [])
+
+    stored_repos = Storage.find_user("hackeryou")
```
`found_user`?

Maikon commented on web/github/user.ex:
```diff
@@ -0,0 +1,13 @@
+    field :repos_url, :string
+
+    timestamps
+
```
Empty line.

gemcfadyen committed:
addressing PR feedback and adding readme

gemcfadyen commented:
@Maikon I've addressed the feedback from the PR.

I've also added a README. Please can you take a look and let me know if I've missed anything.

I'll merge these changes this afternoon. Thanks!
